pandas/_libs/reduction.pyx

```diff
@@ -189,6 +189,24 @@ cdef class _BaseGrouper:
 
         return cached_typ, cached_ityp
 
+    cdef inline object _apply_to_group(self,
```

Comment:
i would return a tuple(res, intitialized) here, less c-like i think is good generally (and this likely doesn't make it differetnt perf wise as this is object anyhow)